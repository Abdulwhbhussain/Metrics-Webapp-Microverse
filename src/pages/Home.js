import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import citiesData from '../redux/citiesData';
import CityCard from '../components/CityCard';
import '../styles/Home.css';

function Home() {
  const [inputSearch, setInputSearch] = useState('');
  const navigate = useNavigate();
  const handleSubmit = (e) => {
    e.preventDefault();
    navigate(`/search/${inputSearch}`);
    setInputSearch('');
  };

  return (
    <div>
      <div className="search-input-container flex justify-center w-[100%] my-10">
        <form onSubmit={handleSubmit} className="flex justify-around w-[90%]">
          <input placeholder="Search City" value={inputSearch} onChange={(e) => setInputSearch(e.target.value)} className=" outline-none border-b-2 text-white border-b-white bg-inherit placeholder:text-white  md:mb-0" />
          <button type="submit" onClick={handleSubmit} className=" md:hidden block hover:bg-sky-700 text-white bg-sky-800 rounded-md py-1 px-3">Search</button>
        </form>
      </div>
      <div className="city-container grid grid-cols-2 md:grid-cols-5 border border-t-white">
        {citiesData.map((city) => <CityCard key={city.name} city={city} />)}
      </div>
    </div>

  );
}
export default Home;
