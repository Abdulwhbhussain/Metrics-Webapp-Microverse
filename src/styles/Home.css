@media screen and (min-width: 375px) {
  .city-container .link-wrapper .city-card {
    padding: 30px 10px;
  }

  .city-container .link-wrapper:nth-child(4n+2),
  .city-container .link-wrapper:nth-child(4n+3) {
    background-color: #2f85c7ef;
  }

  .weather-info li {
    color: whitesmoke;
    font-size: 0.9rem;
  }
}

@media screen and (min-width: 768px) {
  .city-container .link-wrapper .city-card {
    padding: 30px 15px;
  }

  .city-container .link-wrapper:nth-child(odd) {
    background-color: #468bd0e3;
  }

  .city-container .link-wrapper:nth-child(even) {
    background-color: #2f85c7ef;
  }

  .weather-info li {
    font-size: 1rem;
  }
}
