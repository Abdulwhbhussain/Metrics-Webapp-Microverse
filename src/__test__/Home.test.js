import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Home from '../pages/Home';

describe('Home component', () => {
  test('Home component renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Home />
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
