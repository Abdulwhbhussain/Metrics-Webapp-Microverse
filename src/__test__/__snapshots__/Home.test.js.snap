// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Home component renders correctly 1`] = `
<div>
  <div
    className="search-input-container flex justify-center w-[100%] my-10"
  >
    <div
      className="sc-gTRrQi dZDKLG flex justify-around w-[90%]"
      onSubmit={[Function]}
    >
      <input
        className=" outline-none border-b-2 text-white border-b-white bg-inherit placeholder:text-white  md:mb-0"
        onChange={[Function]}
        placeholder="Search City"
        value=""
      />
      <button
        className=" md:hidden block hover:bg-sky-700 text-white bg-sky-800 rounded-md py-1 px-3"
        onClick={[Function]}
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="city-container grid grid-cols-2 md:grid-cols-5 border border-t-white"
    style={Object {}}
  >
    <a
      className=""
      href="/details/Abu Dhabi/24.4667/54.3667"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="abu-dhabi.jpg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Abu Dhabi
            ,
            UAE
          </li>
          <li>
            Latitude:
            24.4667
          </li>
          <li>
            Longitude:
            54.3667
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Amsterdam/52.374/4.8897"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="amsterdam.jpeg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Amsterdam
            ,
            Netherlands
          </li>
          <li>
            Latitude:
            52.374
          </li>
          <li>
            Longitude:
            4.8897
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Bangkok/13.75/100.5167"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="bangkok.jpeg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Bangkok
            ,
            Thailand
          </li>
          <li>
            Latitude:
            13.75
          </li>
          <li>
            Longitude:
            100.5167
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Beijing/39.9062/116.3913"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="beijing.jpeg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Beijing
            ,
            China
          </li>
          <li>
            Latitude:
            39.9062
          </li>
          <li>
            Longitude:
            116.3913
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Berlin/52.517/13.3889"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="berlin.jpg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Berlin
            ,
            Germany
          </li>
          <li>
            Latitude:
            52.517
          </li>
          <li>
            Longitude:
            13.3889
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Cairo/30.0626/31.2497"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="cairo.jpeg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Cairo
            ,
            Egypt
          </li>
          <li>
            Latitude:
            30.0626
          </li>
          <li>
            Longitude:
            31.2497
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Hong Kong/22.2855/114.1577"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="HongKong.jpg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Hong Kong
            ,
            China
          </li>
          <li>
            Latitude:
            22.2855
          </li>
          <li>
            Longitude:
            114.1577
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/London/44.34/10.99"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="London.jpg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            London
            ,
            UK
          </li>
          <li>
            Latitude:
            44.34
          </li>
          <li>
            Longitude:
            10.99
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Moscow/55.7522/37.6156"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="moscow.jpg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Moscow
            ,
            Russia
          </li>
          <li>
            Latitude:
            55.7522
          </li>
          <li>
            Longitude:
            37.6156
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Mumbai/19.0785/72.8782"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="mumbai.jpeg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Mumbai
            ,
            India
          </li>
          <li>
            Latitude:
            19.0785
          </li>
          <li>
            Longitude:
            72.8782
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/New York/39.6853/-93.9269"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="newYork.jpeg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            New York
            ,
            USA
          </li>
          <li>
            Latitude:
            39.6853
          </li>
          <li>
            Longitude:
            -93.9269
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Paris/48.8589/2.32"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="paris.jpeg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Paris
            ,
            France
          </li>
          <li>
            Latitude:
            48.8589
          </li>
          <li>
            Longitude:
            2.32
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Rio De Janeiro/38.3/21.7833"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="Rio.jpeg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Rio De Janeiro
            ,
            Brazil
          </li>
          <li>
            Latitude:
            38.3
          </li>
          <li>
            Longitude:
            21.7833
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Rome/41.8933/12.4829"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="romee.jpeg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Rome
            ,
            Italy
          </li>
          <li>
            Latitude:
            41.8933
          </li>
          <li>
            Longitude:
            12.4829
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Seoul/37.5665/126.978"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="seoul.jpeg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Seoul
            ,
            South Korea
          </li>
          <li>
            Latitude:
            37.5665
          </li>
          <li>
            Longitude:
            126.978
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Singapore/1.2899/103.8519"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="singapore.jpg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Singapore
            ,
            Singapore
          </li>
          <li>
            Latitude:
            1.2899
          </li>
          <li>
            Longitude:
            103.8519
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Spain/40.4167/-3.7036"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="spainn.jpg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Spain
            ,
            Spain
          </li>
          <li>
            Latitude:
            40.4167
          </li>
          <li>
            Longitude:
            -3.7036
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Stockholm/59.3314/18.0594"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="stockholm.jpeg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Stockholm
            ,
            Sweden
          </li>
          <li>
            Latitude:
            59.3314
          </li>
          <li>
            Longitude:
            18.0594
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Tokyo/35.6767/139.764"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="tokyo.jpeg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Tokyo
            ,
            Japan
          </li>
          <li>
            Latitude:
            35.6767
          </li>
          <li>
            Longitude:
            139.764
          </li>
        </ul>
      </div>
    </a>
    <a
      className=""
      href="/details/Yangon/16.8053/96.1561"
      onClick={[Function]}
    >
      <div
        className="sc-bgqQcB hwNKfY card"
      >
        <figure>
          <img
            alt="cloud"
            className=""
            src="yangon.jpg"
            style={Object {}}
          />
        </figure>
        <ul
          className=""
        >
          <li>
            Yangon
            ,
            Myanmar
          </li>
          <li>
            Latitude:
            16.8053
          </li>
          <li>
            Longitude:
            96.1561
          </li>
        </ul>
      </div>
    </a>
  </div>
</div>
`;
